crates/backend/src/codegen.rs

```diff
@@ -1135,6 +1138,135 @@ impl<'a> TryToTokens for ast::Module {
+                use wasm_bindgen::describe::WasmDescribe;
+                use wasm_bindgen::convert::*;
+                use wasm_bindgen::JsValue;
+
+                // interop w/ JsValue
+                impl From<#name> for JsValue {
+                    fn from(val: #name) -> JsValue {
+                        val.obj.into()
+                    }
+                }
+                impl AsRef<JsValue> for #name {
+                    fn as_ref(&self) -> &JsValue { self.obj.as_ref() }
+                }
+                impl AsMut<JsValue> for #name {
+                    fn as_mut(&mut self) -> &mut JsValue { self.obj.as_mut() }
+                }
+
+                // interop w/ Object
+                impl From<#name> for Object {
+                    fn from(val: #name) -> Object {
+                        val.obj
+                    }
+                }
+                impl AsRef<Object> for #name {
+                    fn as_ref(&self) -> &Object { &self.obj }
+                }
+                impl AsMut<Object> for #name {
+                    fn as_mut(&mut self) -> &mut Object { &mut self.obj }
+                }
+
+                // Boundary conversion impls
+                impl WasmDescribe for #name {
+                    fn describe() {
+                        Object::describe();
+                    }
+                }
+
+                impl IntoWasmAbi for #name {
+                    type Abi = <Object as IntoWasmAbi>::Abi;
+                    fn into_abi(self, extra: &mut Stack) -> Self::Abi {
+                        self.obj.into_abi(extra)
+                    }
+                }
+
+                impl<'a> IntoWasmAbi for &'a #name {
+                    type Abi = <&'a Object as IntoWasmAbi>::Abi;
+                    fn into_abi(self, extra: &mut Stack) -> Self::Abi {
+                        (&self.obj).into_abi(extra)
+                    }
+                }
+
+                impl FromWasmAbi for #name {
+                    type Abi = <Object as FromWasmAbi>::Abi;
+                    unsafe fn from_abi(abi: Self::Abi, extra: &mut Stack) -> Self {
+                        #name { obj: Object::from_abi(abi, extra) }
+                    }
+                }
+
+                impl OptionIntoWasmAbi for #name {
+                    fn none() -> Self::Abi { Object::none() }
+                }
+                impl<'a> OptionIntoWasmAbi for &'a #name {
+                    fn none() -> Self::Abi { <&'a Object>::none() }
+                }
+                impl<'a> OptionFromWasmAbi for #name {
+                    fn is_none(abi: &Self::Abi) -> bool { Object::is_none(abi) }
+                }
+            };
+        }).to_tokens(tokens);
+    }
+}
+
+impl ToTokens for ast::DictionaryField {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let name = &self.name;
+        let ty = &self.ty;
+        (quote! {
+            pub fn #name(&mut self, val: #ty) -> &mut Self {
```

Comment:
You use `pub` here, even though the Dictionary might not have a public visibility.